packages/http-client-csharp/generator/Microsoft.Generator.CSharp/test/Helpers.cs

```diff
@@ -36,25 +38,29 @@ private static StackFrame GetRealMethodInvocation(StackTrace stackTrace)
             while (i < stackTrace.FrameCount)
             {
                 var frame = stackTrace.GetFrame(i);
-                if (frame!.GetMethod()!.DeclaringType != typeof(Helpers))
+                var declaringType = frame!.GetMethod()!.DeclaringType!;
+                // we need to skip those method invocations from this class, or from the async state machine when the caller is an async method
+                if (declaringType != typeof(Helpers) && !IsCompilerGenerated(declaringType))
```

Comment:
Shouldn't we just go to the last frame in the stack?

Reply:
The last frame may not work.
One scenario is that this method may be called as a utility in this class, therefore maybe the call stack looks like this:
Real method => GetExpectedFromFile => GetAssetFileOrDirectoryPath (the current method seeing the stack trace)
Therefore the last frame might also be an invocation inside this helper class (I changed this because my new scenario does not want to duplicate this again).
Another scenario that we cannot just take the last frame is that the caller might be an async method. This new condition is checking and unravels those extra stuffs created by the compiler when the caller is async (the async state machine and state machine builder stuff), in order to get the real invoker of this method (which maybe an async test method)

Reply:
I'm probably missing something but I mean the frame at the bottom of the stack which should always be the "real method".